gator/util.py

```diff
@@ -36,10 +36,19 @@ def get_gatorgrader_home():
     # the current_gatorgrader_home is acceptable, so use it
     if verify_gatorgrader_home(current_gatorgrader_home) is not False:
         gatorgrader_home = current_gatorgrader_home
-    # the current GATORGRADER_HOME is not valid, so create the
-    # home for this program to be the current working directory
+    # The current GATORGRADER_HOME is not valid, so create the
+    # home for this program to be the directory from which
+    # GatorGrader is currently run. For instance, when GatorGrader is
```

Comment:
This description makes it seem as though `GATORGRADER_HOME` would be the current working directory, but the example shows it to be the installation's `gator` directory.

Reply:
Should this be elsewhere, perhaps in the module responsible for paths or globals/constants?

Reply:
You are correct about this point. Is should draw attention to the fact that there is a bigger issue at play here: the `GATORGRADER_HOME` variable is now "abused" or "overloaded" to accommodate the way in which `pluginbase` requires the tool to load plugins.

I've thought about this issue a lot and I have some ideas for how to change the code base. With that said, I'm concerned about making the changes in this PR because they will require me to also change the function that calls this function and then test to ensure that `pluginbase` is working correctly in all cases: when run locally without Gradle and locally with Gradle and locally with the test suite and then in the cloud in all of those configurations and then in Docker in the Gradle configurations.

I'm honestly not sure that I can get this exactly right today, given time constraints. Since I'm pretty confident that the program is currently working, is it okay if I make note of this and then agree to raise an issue, refactor the system, test it extensively, and then have a fix in a separate PR?

Please let me know what you think. Thanks!

Reply:
Finally, before I forget, let me make one other point: `util` has become a "dumping ground" for a wide variety of functions that do not fit anywhere else, at least right now. I think that this module needs to be significantly refactored so as to ensure that functions have the correct home. Overall, I think that this will result in many changes to `util`, the addition of new modules, and then adjustments to the modules and functions that called old functions in `util`. As such, this is why I am advocating that we not make this change right now in this PR.

Reply:
I agree with keeping the existing implementation: We may want to change the documentation comment to reflect this conversation.